packages/actions/test/unit/security.test.ts

```diff
@@ -27,57 +45,60 @@ describe("Security", () => {
     const { adminFirestore, adminAuth } = initializeAdminServices()
     const { userApp, userFunctions, userFirestore } = initializeUserServices()
     const userAuth = getAuth(userApp)
-    const user1 = fakeUsersData.fakeUser1
-    const user2 = fakeUsersData.fakeUser2
-    const user3 = fakeUsersData.fakeUser3
-    const user1Pwd = generatePseudoRandomStringOfNumbers(24)
-    const user2Pwd = generatePseudoRandomStringOfNumbers(24)
-    const user3Pwd = generatePseudoRandomStringOfNumbers(24)
-
-    beforeAll(async () => {
-        // create 1st user
-        await createNewFirebaseUserWithEmailAndPw(userApp, user1.data.email, user1Pwd)
-        await sleep(5000)
 
-        // Retrieve the current auth user in Firebase.
-        let currentAuthenticatedUser = getCurrentFirebaseAuthUser(userApp)
-        user1.uid = currentAuthenticatedUser.uid
+    const users = [fakeUsersData.fakeUser1, fakeUsersData.fakeUser2, fakeUsersData.fakeUser3]
+    const passwords = generateUserPasswords(users.length)
 
-        // create 2nd user
-        await createNewFirebaseUserWithEmailAndPw(userApp, user2.data.email, user2Pwd)
-        await sleep(5000)
+    /// @note pre conditions for production tests
+    if (envType === TestingEnvironment.PRODUCTION) {
+        if (
+            !process.env.AUTH_GITHUB_CLIENT_ID ||
+            !process.env.AUTH_USER_EMAIL ||
+            !process.env.AUTH_GITHUB_USER_PW ||
+            !process.env.AUTH_GMAIL_CLIENT_ID ||
+            !process.env.AUTH_GMAIL_CLIENT_SECRET ||
+            !process.env.AUTH_GMAIL_REDIRECT_URL ||
+            !process.env.AUTH_GMAIL_REFRESH_TOKEN
+        )
+            throw new Error("Missing environment variables for Firebase tests.")
+    }
 
-        // Retrieve the current auth user in Firebase.
-        currentAuthenticatedUser = getCurrentFirebaseAuthUser(userApp)
-        user2.uid = currentAuthenticatedUser.uid
-
-        // create the coordinator
-        await createNewFirebaseUserWithEmailAndPw(userApp, user3.data.email, user3Pwd)
-        await sleep(5000)
-
-        currentAuthenticatedUser = getCurrentFirebaseAuthUser(userApp)
-        user3.uid = currentAuthenticatedUser.uid
-
-        // add coordinator privileges
-        await setCustomClaims(adminAuth, user3.uid, { coordinator: true })
+    // create users for all tests
+    beforeAll(async () => {
+        for (let i = 0; i < users.length; i++) {
+            users[i].uid = await createMockUser(
```

Comment:
Assuming this has the same invariants as the previous createNewFirebaseUserWithEmailandPassword ?

Reply:
it's just a wrapper around it to avoid code repetition